backend/server.py

```diff
@@ -0,0 +1,45 @@
+from aiohttp import web
+import socketio
+from whisprer import Whisprer
+import logging
+
+# Configure logging settings
+logging.basicConfig()
+logging.root.setLevel(logging.INFO)
+
+sio = socketio.AsyncServer(cors_allowed_origins=[])
+app = web.Application()
+sio.attach(app)
+
+whisprer = Whisprer()
+
+
+@sio.on("connect")
+def handle_connect(sid, environ):
+    logging.info("Client connected")
+
+
+@sio.on("disconnect")
+def handle_disconnect(sid):
+    logging.info("Disconnection detected")
+    whisprer.disconnect_from_stream(sid)
+
+
+@sio.on("startWhispering")
+async def handle_stream_start(sid, config):
```

Comment:
[minor] I think it makes sense to send the config once, probably one handle_connect when a new client registers

Reply:
The reason was to increase readability when emitting events over the socket to the server from the client.

Before the client can start recording, it needs to ensure two things:

- The socket is established.
- The server is ready to receive audio chunks.

It can take some time to initialize the stream in the server (loading the Whisper model and the diarization pipeline, starting the threads, etc), so the server is configured to emit a "whisperingStarted" event once it's ready to receive audio chunks.

The client emits a "startWhispering" event to the server along with the transcription config once it's connected to the socket, and the "whisperingStarted" event sent by the server directly corresponds to it. So it's mostly for readability and individualizing the connection stage and initialization stage.

It's not strictly necessary, as all this can be done directly when connecting to the socket.

What do you think?

Reply:
I think it makes more sense to share the configuration as part of the initialization phase